sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/metadata/ContainerRequestContextHolder.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.metadata;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.PreMatching;
+import org.slf4j.MDC;
+
+/**
+ * A filter that provides the current request from a {@link ThreadLocal}. Currently, it is only
+ * implemented for the client filters in this package.
+ */
+@Priority(Priorities.USER)
+@PreMatching
+public class ContainerRequestContextHolder
+    implements ContainerRequestFilter, ContainerResponseFilter {
+
+  private static final ThreadLocal<ContainerRequestContext> REQUEST_CONTEXT_HOLDER =
+      new ThreadLocal<>();
+
+  /**
+   * @return The current {@link ContainerRequestContext} if the current {@link Thread} is in a
+   *     request context. An empty {@code Optional} if the current thread is not in a request
+   *     context.
+   */
+  static Optional<ContainerRequestContext> currentRequestContext() {
+    return Optional.ofNullable(REQUEST_CONTEXT_HOLDER.get());
+  }
```

Comment:
Can't we use Spring's [RequestContextHolder](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/context/request/RequestContextHolder.html)?

Reply:
Imho this is not needed in the Spring context. The important part of transferring the metadata context should be included in the [ContextCopyTaskDecorator](https://github.com/SDA-SE/sda-spring-boot-commons/blob/master/sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/async/ContextCopyTaskDecorator.java). As @christopher-cudennec said, Spring's RequestContextHolder should be already doing what's implemented here.